src/definitions.ts

```diff
@@ -5,9 +5,39 @@ declare module "@capacitor/core" {
 }
 
 export interface SignInWithApplePlugin {
+  Init(options: InitOptions): Promise<void>;
+
   Authorize(): Promise<ResponseSignInWithApplePlugin>;
 }
 
+export interface InitOptions {
+  clientId: string;
+  scope: string;
+  redirectURI: string;
+  state: string;
+  usePopup: boolean;
+}
+
+export interface SignInError {
+  error: string;
+}
+
+export interface Authorization {
+  code: string;
+  id_token: string;
+  state: string;
+}
+
+export interface User {
+  email: string;
+  name: Name;
+}
+
+export interface Name {
+  firstName: string;
+  lastName: string;
+}
+
 export interface ResponseSignInWithApplePlugin {
```

Comment:
I'm not sure this is a good approach - aggregating the types and nesting like that seems quite unnecessary when you can specify multiple types that could be returned?

If you're keen to have only one type that could be returned, then it would be better to adjust either the iOS or Web plugin response payload structures to match the other, no?

Reply:
48a795d fixed it